### PR TITLE
gh-2875 fix workflow reporting on push events

### LINKED ISSUE #2875: E2E tests failing at publish-results job
Unfortunately, merging the pull req didn't fix the E2E failures:
https://github.com/hawtio/hawtio/actions/runs/5452977928

We need to find and fix the root cause of why it happens. @mmuzikar Could you have a look?

### PR DESCRIPTION
**Pull request by mmuzikar**
Closes #2875

### CODE REVIEW

**Review by tadayosi — APPROVED** [review]

**Comment by tadayosi** [comment]
@mmuzikar Looks like it stil fails at the last stage.

**Comment by mmuzikar** [comment]
Yes - I'm looking at it, however it doesn't fail on my fork so I have to debug it here :/

**Comment by tadayosi** [comment]
So, it could be related to the permissions on the hawtio project.

File: .github/workflows/e2e_test.yml [hunk 1]
```diff
@@ -112,9 +112,7 @@ jobs:
           key: e2e-docker-images-${{ matrix.java }}-${{ github.run_id }}-${{ github.run_attempt }}
 
   publish-results:
-    permissions:
-      checks: write
-      pull-requests: write
+    permissions: write-all
```
Comment (tadayosi) [inline .github/workflows/e2e_test.yml:115]: Should we still need this?  Can't we revert to the original?  The less permissive, the better.

**Comment by tadayosi** [comment]
Could be this?
https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow
> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

**Comment by tadayosi** [comment]
So, how should we proceed with this?

**Comment by mmuzikar** [comment]
This change should now have access to the secrets and checkout the correct branch... but due to how the `pull_request_target` works the tests won't run in this PR

**Comment by mmuzikar** [comment]
@tadayosi here you can see the workflow working: https://github.com/hawtio/hawtio/pull/2889 and push works as well: https://github.com/hawtio/hawtio/actions/runs/5575141223

**Comment by tadayosi** [comment]
OK, let's merge then.

**Pull request merged**